src/main/java/io/xstefank/wildfly/bot/ConfigFileChangeProcessor.java

```diff
@@ -46,14 +50,20 @@ void onFileChanged(@PullRequest.Opened @PullRequest.Edited @PullRequest.Synchron
                     Optional<WildFlyConfigFile> file = Optional.ofNullable(yamlObjectMapper.readValue(updatedFileContent, WildFlyConfigFile.class));
 
                     if (file.isPresent()) {
-                        githubCommitProcessor.commitStatusSuccess(pullRequest, CHECK_NAME, "Valid");
-                        Log.debug("Configuration File check successful");
+                        List<String> problems = validateFile(file.get());
+                        if (problems.isEmpty()) {
+                            githubCommitProcessor.commitStatusSuccess(pullRequest, CHECK_NAME, "Valid");
+                            Log.debug("Configuration File check successful");
+                        } else {
+                            githubCommitProcessor.commitStatusError(pullRequest, CHECK_NAME, "Rule is missing an id or multiple rules have the same id.");
```

Comment:
why not pass problems into the status too?

Reply:
It would be unreadable in the status info I would assume. That's why it sends only recommendation. We might extract which problems are present form the return `List` and log only relevant problems.

Reply:
even a string this long won't fit into the status window and that is expected. If you hover your mouse over the text, it shows it in full.